File: AtCoder/ACP/BootCamp4b/Hard100/python/B_Robot_Arms.py
N = int(input())
V = []
for i in range(N):
    X, L = [int(x) for x in input().split()]
    V.append([X + L, X - L])
V.sort()
end = V[0][0]
cnt = 1
for v in V:
    if end <= v[1]:
        cnt += 1
        end = v[0]
print(cnt)

File: AtCoder/ACP/BootCamp4b/Hard100/cpp/B_Robot_Arms.cpp
#include <algorithm>
#include <iostream>
#include <stdio.h>
#include <vector>

using namespace std;

int main() {
  int N;
  cin >> N;
  vector<vector<int>> V(N);
  for (int i = 0; i < N; ++i) {
    int X, L;
    cin >> X >> L;
    V[i] = vector<int>{X + L, X - L};
  }
  sort(V.begin(), V.end());
  int end = V[0][0];
  int cnt = 1;
  for (int i = 1; i < N; ++i) {
    if (end <= V[i][1]) {
      ++cnt;
      end = V[i][0];
    }
  }
  printf("%d\n", cnt);
}